src/backend/distributed/planner/local_distributed_join_planner.c

```diff
@@ -0,0 +1,493 @@
+#include "postgres.h"
+
+#include "distributed/pg_version_constants.h"
+
+#include "funcapi.h"
+
+#include "catalog/pg_type.h"
+#include "catalog/pg_class.h"
+#include "catalog/pg_index.h"
+#include "distributed/citus_nodes.h"
+#include "distributed/citus_ruleutils.h"
+#include "distributed/commands.h"
+#include "distributed/commands/multi_copy.h"
+#include "distributed/distributed_planner.h"
+#include "distributed/errormessage.h"
+#include "distributed/local_distributed_join_planner.h"
+#include "distributed/listutils.h"
+#include "distributed/log_utils.h"
+#include "distributed/metadata_cache.h"
+#include "distributed/multi_logical_planner.h"
+#include "distributed/multi_logical_optimizer.h"
+#include "distributed/multi_router_planner.h"
+#include "distributed/multi_physical_planner.h"
+#include "distributed/multi_server_executor.h"
+#include "distributed/query_colocation_checker.h"
+#include "distributed/query_pushdown_planning.h"
+#include "distributed/recursive_planning.h"
+#include "distributed/relation_restriction_equivalence.h"
+#include "distributed/log_utils.h"
+#include "distributed/shard_pruning.h"
+#include "distributed/version_compat.h"
+#include "lib/stringinfo.h"
+#include "optimizer/clauses.h"
+#if PG_VERSION_NUM >= PG_VERSION_12
+#include "optimizer/optimizer.h"
+#else
+#include "optimizer/var.h"
+#endif
+#include "optimizer/planner.h"
+#include "optimizer/prep.h"
+#include "parser/parsetree.h"
+#include "nodes/makefuncs.h"
+#include "nodes/nodeFuncs.h"
+#include "nodes/nodes.h"
+#include "nodes/nodeFuncs.h"
+#include "nodes/pg_list.h"
+#include "nodes/primnodes.h"
+#if PG_VERSION_NUM >= PG_VERSION_12
+#include "nodes/pathnodes.h"
+#else
+#include "nodes/relation.h"
+#endif
+#include "utils/builtins.h"
+#include "utils/guc.h"
+#include "utils/lsyscache.h"
+
+typedef struct RangeTableEntryDetails
+{
+	RangeTblEntry *rangeTableEntry;
+	Index rteIndex;
+	List *restrictionList;
+	List *requiredAttributeNumbers;
+	bool hasConstantFilterOnUniqueColumn;
+} RangeTableEntryDetails;
+
+typedef struct ConversionCandidates
```

Comment:
Umm it seems that conversion candidates and resultRelationId were both passed to `GetNextRTEToConvertToSubquery`, however we were not using `resultRelationId` there, so I removed it. I guess now putting it inside this struct will not be helpful because there isn't any other common place where we pass both of them now